src/inmanta/server/services/resourceservice.py

```diff
@@ -426,6 +426,7 @@ async def _update_deploy_state(
         message: str,
         fail_on_error: bool,
         connection: Optional[Connection] = None,
+        can_overwrite_available: bool = True,
```

Comment:
Just a question: Why did we parameterize this method. As far as I can see we never use `can_overwrite_available: bool = True` and I don't think we have the intention of ever using it like that.

Reply:
True, I kept it at one hand to get this out quickly, at the other hand I use it as documentation for the unexpected behavior that it will not always update the state.